resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1428,6 +1428,12 @@ private object Client extends Logging {
       return false
     }
 
+    val srcAuthority = srcUri.getAuthority()
+    val dstAuthority = dstUri.getAuthority()
+    if (srcAuthority != null && !srcAuthority.equalsIgnoreCase(dstAuthority)) {
```

Comment:
Now, if src is null and dst is not, this fails to detect they're different. Please see my previous comment again.

Reply:
This very method uses `Objects.equal` to solve that problem for hosts below.